### PR TITLE
Fixed yarn clean script for Windows OS

### PR DESCRIPTION
**Pull request by przemyslaw-zan**
Fix (utils): The `tools.clean()` function will resolve paths correctly on Windows environments. Closes ckeditor/ckeditor5#10141.

### CODE REVIEW

File: packages/ckeditor5-dev-utils/tests/tools.js [hunk 1]
```diff
@@ -577,6 +577,17 @@ describe( 'utils', () => {
...
 			} );
+
+			it( 'works with paths that contain backslash', () => {
+				return tools.clean( 'test\\path', '**\\packages' )
+					.then( () => {
+						expect( delArg ).to.equal( path.join( 'test/path', '**/packages' ) );
+						expect( loggerVerbosity ).to.equal( 'info' );
+						expect( infoSpy.calledTwice ).to.equal( true );
+						expect( infoSpy.firstCall.args[ 0 ] ).to.match( new RegExp( files[ 0 ] ) );
+						expect( infoSpy.secondCall.args[ 0 ] ).to.match( new RegExp( files[ 1 ] ) );
```
Comment (psmyrek) [inline packages/ckeditor5-dev-utils/tests/tools.js:588]: These lines are not related to the subject of this test case - you are testing here if the joined path, that is sent to the `del` package, contains only slashes, but the `infoSpy` still operates on the `files` array that does not match the provided path `'test/path/**/packages'` at all. So comparing the `infoSpy` args with the `files` when the joined path does not match them is confusing.

**Review by psmyrek — CHANGES_REQUESTED** [review]
It seems that it works fine 🎉.

I've just found one minor thing to change.

File: packages/ckeditor5-dev-utils/tests/tools.js (hunk 1 above)
Comment (przemyslaw-zan) [reply packages/ckeditor5-dev-utils/tests/tools.js]: Addresed

**Review by psmyrek — APPROVED** [review]
LGTM 🎉
